### DISCUSSION TITLE
tax_filter() before tax_agg()?

### DISCUSSION THREAD

**Discussion by MSMortensen** [david-barnett/microViz#111] [General]
Hi David,

First of all I want to thank you for the work you have put into this package.
It is very useful and intuitive to use.

One thing I would like to discuss is not the actual functions, but your suggested application of them, especially the use of `tax_filter()` prior to using `tax_agg()`.

Would it not be better to agglomerate before filtering?

If filtering at genus level and then agglomerating at family level it is very likely that some families will be underrepresented as some of their genera has been removed during filtering.

Kind regards,
Martin

PS: I liked your presentation at MVIF today.

**Comment by david-barnett** [comment]
Hi Martin

Thanks for your kind words 😊

I agree with your reasoning, and think this is a nice discussion point to raise here. I feel the range of choices one can make in filtering taxa before analysis is an underappreciated problem. In some datasets (but thankfully not very many in my experience so far) alternative choices in handling rarer taxa may have surprisingly large impacts on downstream analyses.

From a practical perspective, tax_filter is capable of performing the filtering at a higher rank, e.g. family, and returning the original level of detail in classification, e.g. ASVs, by setting the tax_level argument. 

In that way, you can make the filtering and aggregation choices more independently of each other, as appropriate for each analysis. 

What exactly "appropriate" is for a given analysis is a bit of an open question I think. But in any case (for anyone else reading this discussion later) always document your filtering choices precisely in your methods section.

In additions to microViz later this year, I hope to add more discussion about filtering, and maybe an interactive tool to help assess the impact of different thresholds and other choices.

Does this help?

Cheers,
David
